Adding xSqlAlias resource

Matticusau opened the pull request.

Matticusau committed:
Removing of tab characters

msftclas commented:
Hi __@Matticusau__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, MSBOT;

Matticusau committed:
Updated xSqlServerInstall for best practices

Updated xSqlServerInstall for better alignment with best practices.
This includes
- Separation of file locations
- Control of Collocation
- Specification of service accounts

Matticusau committed:
Revert "Updated xSqlServerInstall for best practices"

This reverts commit bab990d4f18c87f6830e9971b3c46df578062021.

Matticusau committed:
Revert "Removing of tab characters"

This reverts commit afb075b302f3e2e99faf7a48aec688e6706b7ca2.

Matticusau committed:
Updated xSqlServerInstall for best practices

Updated xSqlServerInstall for best practices including
- Separation of directories
- Specification of service accounts
- Control of collation

Matticusau committed:
Updated resource version and readme

Matticusau committed:
Updated xSqlServerInstall for best practices pt2

Updated xSqlServerInstall for best practices as first submission failed
Includes
- Ability to separate paths
- Support for multiple install media versions
- Control of collation

Matticusau commented on DSCResources/MSFT_xSqlServerInstall/MSFT_xSqlServerInstall.psm1:
```diff
@@ -69,6 +77,8 @@ function Set-TargetResource
 
         [PSCredential] $SourcePathCredential,
+        
+        [string]$VersionID="120",
```
@TravisEz13 Can you please explain your comment further? the $VersionID parameter is only defined once in each of the "Test-", "Set-", and "Get-" functions. It has a default value assigned so that it will work with the latest release by default. I am not sure why you are requesting that it be defined in a common location.

Matticusau committed:
Update xSqlPs.psd1

Reverted the version change I made to comply with the guidelines

Matticusau committed:
Added the DefaultVersionID global variable

The DefaultVersionID will determine the value for the VersionID property
when not supplied. By specifying this in a single location this will
simplify the process of changing the default version of SQL i.e. when
new releases are made so we always support the latest by default.

Matticusau committed:
Added xSQLAlias resource

Added a new resource to manage SQL Aliases. This is useful in
environments where applications have dependencies on particular server
names but SQL consolidations/migrations have occurred. It handles both
native and wow6432node registry locations which is a common oversight in
configuring a Client Alias.

Matticusau committed:
Added help for xSqlAlias to README.md

Added help info for xSqlAlias

Matticusau committed:
Slight change to xSqlAlias help

Matticusau committed:
Minor change to MOF

Matticusau committed:
Deleting MFST_xSQLAlias.psm1 to update file name

Matticusau committed:
Deleting MFST_xSQLAlias.schema.mof to update file name

Matticusau committed:
Added xSqlAlias resource

Added a new resource to manage Client Aliases. Handles both native and
wow6432bit registry paths. Handy for configuring servers which host
applications that have dependencies on certain SQL Server names.
This time it was submitted with files aligning to the naming convention
of 'Sql' not all uppercase

Matticusau committed:
Added header comment to file

Matticusau commented on DSCResources/MSFT_xSqlServerInstall/MSFT_xSqlServerInstall.schema.mof:
```diff
@@ -19,5 +21,10 @@ class MSFT_xSqlServerInstall : OMI_BaseResource
     [write, Description("Specify the startup account for the SQL server service.")] string SvcAccount;
     [write, Description("Specify logins to be members of the sysadmin role.")] string SysAdminAccounts;
     [write, Description("Specify the account for SQL server agent service.")] string AgentSvcAccount;
+    [write, Description("Specify the default collation for SQL server")] string SqlCollation;
```
I thought about adding a Value Map but do you think it is wise given the number of collations available in SQL (across languages)?

Matticusau committed:
Removing MSFT_xSqlAlias.psm1 for reload

Removing MSFT_xSqlAlias.psm1 for reload in no binary encoding

Matticusau committed:
Removing MSFT_xSqlAlias.schema.mof for reload

Removing MSFT_xSqlAlias.schema.mof for reload in no binary encoding

Matticusau committed:
Minor changes based on github comments

Matticusau committed:
Recommit of xSqlAlias

Recommit of xSqlAlias to remove binary formating

Matticusau committed:
Re-encoded to UTF-8

Matticusau committed:
Added Pester Test for xSqlAlias

As requested in comments from last submission I have added a tests file
for xSqlAlias for use with Pester. It has been tested locally on my
machine.

Matticusau committed:
Removed schema check from xSqlAlias tests file

Had an issue with the detection of xSqlAlias resource by get-dscresource
in Pester testing. Removing schema checks for now as these weren't
critical at this time.

Matticusau committed:
Merge remote-tracking branch 'refs/remotes/PowerShell/dev' into dev

Matticusau commented:
Hi @TravisEz13 just wondering if there is anything else required for this pull request? I cannot see any issues and have merged it with other changes that have been rolled out since.

KarolKaczmarek commented on README.md:
```diff
@@ -68,6 +81,14 @@ Each database can belong to only one HA group.
+* **TCPPort**: The tcp port of the instance.
+
 ## Versions
 
```
Add section

### Unreleased
- Added new resource xSqlAlias

Matticusau replied:
Thanks
I have added the note to the readme.md based on what I could see in other DSC resources. If there is anything else please let me know

Matticusau committed:
Merge remote-tracking branch 'refs/remotes/PowerShell/dev' into dev

Matticusau committed:
Added unreleased help note and bug fixes

Added the requested Unreleased note in the readme.md
Also fixed some bugs regarding the directory properties on the
xSQLServerInstall resource as reported in my private repository. Paths
with spaces are now supported.
